### PR TITLE
abseil-cpp: 20200923.3 -> 20210324.1

### PR DESCRIPTION
**Pull request by gebner**
###### Motivation for this change

The bear build was broken due to grpc upgrade in #123855.  Upgrading abseil seemed liked the easiest fix.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Added a release notes entry if the change is major or breaking
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/libraries/abseil-cpp/default.nix [hunk 1]
```diff
@@ -2,15 +2,21 @@
...
     repo = "abseil-cpp";
     rev = version;
-    sha256 = "1p4djhm1f011ficbjjxx3n8428p8481p20j4glpaawnpsi362hkl";
+    sha256 = "16w63brfwgiayiyhvawsnr14xyy5hpp68k8fj0z6yk0bjzw6jvjw";
   };
 
+  patches = [
+    # Use CMAKE_INSTALL_FULL_{LIBDIR,INCLUDEDIR}
+    # https://github.com/abseil/abseil-cpp/pull/963
+    ./cmake-full-dirs.patch
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/abseil-cpp/default.nix:17]: Can we fetch the patch instead?

File: pkgs/development/libraries/science/math/or-tools/default.nix [hunk 2]
```diff
@@ -17,25 +17,16 @@
...
     repo = "or-tools";
     rev = "v${version}";
-    sha256 = "1zqgvkaw5vf2d8pwsa34g9jysbpiwplzxc8jyy8kdbzmj8ax3gpg";
+    sha256 = "0yihrsg8wj4b82xwg1hbn97my8zqd7xhw7dk7wm2axsyvqd6m3b3";
   };
 
-  patches = [
-    # This patch (on master as of Feb 11, 2021) fixes or-tools failing to respect
-    # USE_SCIP=OFF and then failing to find scip/scip.h
-    (fetchpatch {
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/science/math/or-tools/default.nix:33]: Please remove the unused fetchpatch from inputs.

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 124030` at c4b5c5c7 run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>44 packages marked as broken and skipped:</summary>
  <ul>
    <li>python38Packages.arviz</li>
    <li>python38Packages.dm-sonnet</li>
    <li>python38Packages.graph_nets</li>
    <li>python38Packages.optuna</li>
    <li>python38Packages.pymc3</li>
    <li>python38Packages.rl-coach</li>
    <li>python38Packages.spacy</li>
    <li>python38Packages.spacy-transformers</li>
    <li>python38Packages.tensorflow-probability</li>
    <li>python38Packages.tensorflowWithCuda</li>
    <li>...</li>
  </ul>
</details>
<details>
  <summary>28 packages skipped due to time constraints:</summary>
  <ul>
    <li>bear</li>
    <li>deja-dup</li>
    <li>duplicity</li>
    <li>duply</li>
    <li>fcitx5-mozc</li>
    <li>home-assistant</li>
    <li>kmymoney</li>
    <li>murmur</li>
    <li>or-tools</li>
    <li>python38Packages.baselines</li>
    <li>...</li>
  </ul>
</details>
<details>
  <summary>89 packages built successfully:</summary>
  <ul>
    <li>abseil-cpp</li>
    <li>beancount (python38Packages.beancount)</li>
    <li>dvc-with-remotes</li>
    <li>fava</li>
    <li>gcalcli</li>
    <li>git-annex-remote-googledrive</li>
    <li>grpc</li>
    <li>ledger2beancount</li>
    <li>lieer</li>
    <li>lndmanage</li>
    <li>python38Packages.beancount_docverif</li>
    <li>python38Packages.chirpstack-api</li>
    <li>python38Packages.drivelib</li>
    <li>python38Packages.goobook</li>
    <li>python38Packages.google-api-core</li>
    <li>python38Packages.google-api-python-client</li>
    <li>python38Packages.google-cloud-access-context-manager</li>
    <li>python38Packages.google-cloud-asset</li>
    <li>python38Packages.google-cloud-automl</li>
    <li>python38Packages.google-cloud-bigquery</li>
    <li>python38Packages.google-cloud-bigquery-datatransfer</li>
    <li>python38Packages.google-cloud-bigtable</li>
    <li>python38Packages.google-cloud-container</li>
    <li>python38Packages.google-cloud-core</li>
    <li>python38Packages.google-cloud-dataproc</li>
    <li>python38Packages.google-cloud-datastore</li>
    <li>python38Packages.google-cloud-dlp</li>
    <li>python38Packages.google-cloud-dns</li>
    <li>python38Packages.google-cloud-error-reporting</li>
    <li>python38Packages.google-cloud-firestore</li>
    <li>python38Packages.google-cloud-iam</li>
    <li>python38Packages.google-cloud-iot</li>
    <li>python38Packages.google-cloud-kms</li>
    <li>python38Packages.google-cloud-language</li>
    <li>python38Packages.google-cloud-logging</li>
    <li>python38Packages.google-cloud-monitoring</li>
    <li>python38Packages.google-cloud-org-policy</li>
    <li>python38Packages.google-cloud-os-config</li>
    <li>python38Packages.google-cloud-pubsub</li>
    <li>python38Packages.google-cloud-redis</li>
    <li>python38Packages.google-cloud-resource-manager</li>
    <li>python38Packages.google-cloud-runtimeconfig</li>
    <li>python38Packages.google-cloud-secret-manager</li>
    <li>python38Packages.google-cloud-securitycenter</li>
    <li>python38Packages.google-cloud-spanner</li>
    <li>python38Packages.google-cloud-speech</li>
    <li>python38Packages.google-cloud-storage</li>
    <li>python38Packages.google-cloud-tasks</li>
    <li>python38Packages.google-cloud-texttospeech</li>
    <li>python38Packages.google-cloud-trace</li>
    <li>python38Packages.google-cloud-translate</li>
    <li>python38Packages.google-cloud-videointelligence</li>
    <li>python38Packages.google-cloud-vision</li>
    <li>python38Packages.google-cloud-websecurityscanner</li>
    <li>python38Packages.googleapis-common-protos</li>
    <li>python38Packages.grpc_google_iam_v1</li>
    <li>python38Packages.ortools</li>
    <li>python38Packages.pathy</li>
    <li>python38Packages.proto-plus</li>
    <li>python38Packages.pydrive</li>
    <li>python38Packages.weboob</li>
    <li>python38Packages.woob</li>
    <li>python39Packages.beancount</li>
    <li>python39Packages.beancount_docverif</li>
    <li>python39Packages.chirpstack-api</li>
    <li>python39Packages.drivelib</li>
    <li>python39Packages.goobook</li>
    <li>python39Packages.google-api-core</li>
    <li>python39Packages.google-api-python-client</li>
    <li>python39Packages.google-cloud-access-context-manager</li>
    <li>python39Packages.google-cloud-bigtable</li>
    <li>python39Packages.google-cloud-core</li>
    <li>python39Packages.google-cloud-dns</li>
    <li>python39Packages.google-cloud-firestore</li>
    <li>python39Packages.google-cloud-logging</li>
    <li>python39Packages.google-cloud-org-policy</li>
    <li>python39Packages.google-cloud-resource-manager</li>
    <li>python39Packages.google-cloud-runtimeconfig</li>
    <li>python39Packages.google-cloud-trace</li>
    <li>python39Packages.google-cloud-videointelligence</li>
    <li>python39Packages.googleapis-common-protos</li>
    <li>python39Packages.grpc_google_iam_v1</li>
    <li>python39Packages.oauthenticator</li>
    <li>python39Packages.ortools</li>
    <li>python39Packages.proto-plus</li>
    <li>python39Packages.pydrive</li>
    <li>python39Packages.weboob</li>
    <li>python39Packages.woob</li>
    <li>wal_e</li>
  </ul>
</details>
<details>
  <summary>1 suggestion:</summary>
  <ul>
    <li>warning: unused-argument

Unused argument: `fetchpatch`.
Near pkgs/development/libraries/science/math/or-tools/default.nix:4:3:
```
  |
4 | , fetchpatch
  |   ^
```
</li>
  </ul>
</details>

**Comment by gebner** [comment]
I can't reproduce the build error on ofborg (apparently some dependency of rippled fails to build), rippled builds fine here.

**Comment by gebner** [comment]
Merging this PR since it also fixes other packages like murmur.  https://github.com/NixOS/nixpkgs/pull/123855#issuecomment-846438125

**Pull request merged**